--- ChangeLog: In timespec.h casted the int to signed long long int
--- 	       	  in timespec_cmp(...) to prevent integer overflow.
--- timespec.h	2014-11-08 16:05:29.450223000 -0800
+++ newtimespec.h	2014-11-08 16:04:43.610257000 -0800
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  signed long long int diff = (signed long long int) a.tv_sec - (signed long long int) b.tv_sec;
+  if (diff < 0)
+    return -1;
+  if (diff == 0)
+    return 0;
+  else //diff >O
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
